ironfish/src/memPool/feeEstimator.ts

```diff
@@ -139,6 +146,58 @@ export class FeeEstimator {
     return this.queue.length
   }
 
+  async estimateFee(
+    percentile: number,
+    sender: Account,
+    receives: { publicAddress: string; amount: bigint; memo: string }[],
+  ): Promise<bigint> {
+    const estimateFeeRate = this.estimateFeeRate(percentile)
+    const estimateTransactionSize = await this.getPendingTransactionSize(
+      sender,
+      receives,
+      estimateFeeRate,
+    )
+    return estimateFeeRate * BigInt(estimateTransactionSize)
+  }
+
+  private async getPendingTransactionSize(
+    sender: Account,
+    receives: { publicAddress: string; amount: bigint; memo: string }[],
+    estimateFeeRate?: bigint,
+  ): Promise<number> {
+    let size = 0
+    size += 8 // spends length
+    size += 8 // notes length
+    size += 8 // fee
+    size += 4 // expiration
+    size += 64 // signature
+
+    const amountNeeded = receives.reduce((acc, receive) => acc + receive.amount, BigInt(0))
+
+    const { amount, notesToSpend } = await this.wallet.createSpends(sender, amountNeeded)
+
+    size += notesToSpend.length * SPEND_SERIALIZED_SIZE_IN_BYTE
+
+    size += receives.length * NOTE_SERIALIZED_SIZE_IN_BYTE
+
+    if (estimateFeeRate) {
+      const additionalAmountNeeded =
+        estimateFeeRate * BigInt(Math.ceil(size / 1000)) - (amount - amountNeeded)
+
+      if (additionalAmountNeeded > 0) {
+        const { notesToSpend: additionalNotesToSpend } = await this.wallet.createSpends(
+          sender,
+          additionalAmountNeeded,
+        )
```

Comment:
@ygao76 I realized there's a problem with this second call to `createSpends` -- it will iterate over the same notes that have already been included in `notesToSpend`